File: 2572-append-characters-to-string-to-make-subsequence/2572-append-characters-to-string-to-make-subsequence.py
class Solution:
    def appendCharacters(self, s: str, t: str) -> int:
        
        n = len(t)
        m = len(s)

        i = j = 0

        while i < n and j < m:
            print(f"i: {i} and j: {j}")
            if s[j] == t[i]:
                i += 1
                j += 1
            else:
                j += 1

        

        return n - i

File: 2572-append-characters-to-string-to-make-subsequence/2572-append-characters-to-string-to-make-subsequence.js
/**
 * @param {string} s
 * @param {string} t
 * @return {number}
 */
var appendCharacters = function(s, t) {
    
    let n = t.length;
    let m = s.length;

    let i = 0;
    let j = 0;

    while (i < n && j < m){
        if (t[i] == s[j]){
            i++;
            j++
        }
        else{
            j++
        }
    }

    return n - i
};